mitreattack/navlayers/generators/usage_generator.py

```diff
@@ -37,113 +45,55 @@ def __init__(self, source, matrix='enterprise', local=None):
             print(f"[UsageGenerator] - unable to load collection {matrix} (current source = {source}).")
             raise BadInput
 
-    def remove_revoked(self, listing):
-        """
-        Remove revoked elements from the listing
-        :param listing: input element list
-        :return: input element list - revoked elements
-        """
-        removed = []
-        for x in listing:
-            if 'revoked' in x:
-                if x['revoked']:
-                    removed.append(x)
-        return [a for a in listing if a not in removed]
-
-    def get_group(self, match):
-        """
-        Retrieve group for a given match
-        :param match: matching pattern
-        :return: first matching group or UnableToFindGroup exception
-        """
-        a = []
-        if match.startswith('G'):
-            filts = [Filter('type', '=', 'intrusion-set')]
-            temp = self.source_handle.query(filts)
-            for b in temp:
-                for c in b['external_references']:
-                    if c['source_name'] == "mitre-attack" and c['external_id'] == match:
-                        return a
-        else:
-            filts = [Filter('type', '=', 'intrusion-set'),
-                     Filter('aliases', '=', match)]
-            a = self.source_handle.query(filts)
-        if len(a):
-            return a[0]
-        else:
-            raise UnableToFindGroup
-
-    def get_software(self, match):
+    def get_stix_object(self, match):
         """
-        Retrieve software for a given match
-        :param match: matching pattern
-        :return: first matching software or UnableToFindSoftware exception
+        Retrieve the stix object for a given string
+        :param match: The string to match on - can be a name, alias, or ATT&CK ID
+        :return: the corresponding stix object
         """
         filts = [
-            [Filter('type', '=', 'malware')],
-            [Filter('type', '=', 'tool')]
+            [Filter('name', '=', match)],
+            [Filter(match, 'in', 'aliases')],
+            [Filter(match, 'in', 'x_mitre_aliases')],
+            [Filter('external_references.external_id', '=', match)],
         ]
-        all_software = list(chain.from_iterable(self.source_handle.query(f) for f in filts))
-        for x in all_software:
-            for y in x['external_references']:
-                if y['source_name'] == 'mitre-attack' and y['external_id'] == match:
-                    return x
-            if match == x['name']:
-                return x
-            if 'aliases' in x:
-                if match in x['aliases']:
-                    return x
-        raise UnableToFindSoftware
+        data = list(chain.from_iterable(self.source_handle.query(f) for f in filts))
+        data = remove_revoked(data)
+        if len(data) == 1:
+            return data[0]
```

Comment:
What happens if there are 2+ matches? That would only happen in weird edge cases (overlapping aliases for instance) but we may want to handle that case with a sensible error, especially for the CLI.

Reply:
In stock ATT&CK data it wouldn't ever happen unless someone made a big mistake and created a duplicate object, but you never know about custom input data.

Reply:
Overall I really like this method of having a single getter function though.

Reply:
I don't think a full on error is necessarily the best course of action here, but I've put in a warning message that prints if this ever occurs. If you want it to be a full on error, let me know and I can change it to that.